18F/identity-idp
# LG-10147 Redirect After Try again when resolution fails

## 🎫 Ticket
[LG-10147](https://cm-jira.usa.gov/browse/LG-10147) FMS/Verify Redirect to VerifyInfo After Try Again

## 🛠 Summary of changes
* Passed in flow of 'in_person' when on the In Person Proofing flow to be able to hit one of the idv_in_person urls.
* Added an additional evaluation to check the feature flag
- flag is on: direct to /verify_info

## Acceptance Criteria 

- When the `in_person_verify_info_controller_enabled` is set to true, clicking Try again on `/verify/session/errors/warning` will take you to `/verify/in_person/verify_info` rather than `/verify/document_capture` 

## 📜 Testing Plan

1. Set `in_person_verify_info_controller_enabled` to true
- [x] Run through steps described below so that resolution fails. Check redirect is working as expected.
- [x] Run through flow with a good SSN so that resolution does not fail. Ensure you can make it through flow to get a barcode

2. Set `in_person_verify_info_controller_enabled` to false. 
* Url should be `/verify/in_person/verify` rather than `/verify/in_person/verify_info`
* Flow with feature flag disabled should not be affected. This is to check that existing functionality did not break.
- [x] Run through steps described below so that resolution fails. Check redirect is working as expected.
- [x] Run through flow with a good SSN so that resolution does not fail. Ensure you can make it through flow to get a barcode
     
3. Run through Team Ada's flow- Successfully verify your identity on-line
The non-in person proofing flow should not be affected. This is to check that existing functionality did not break.
- [x] Run through team Ada's flow. Enter a bad SSN to fail resolution. Check redirect is working as expected.
- [x] Run through flow with a good SSN so that resolution does not fail. Ensure you can make it through flow to get a barcode
 
##  Checklist of steps to confirm the changes
1. Create an account
3. Attempt to verify your identify by uploading yml files to fail identify verification
5. Select In Person Flow, search and select a post office
6. Fill in State Id information
7. **Once on Enter your Social Security Number use '901' rather than '900' or '666' to fail  ie: 901-xxx-xxxx**
8. Click Continue on `/verify/in_person/verify_info`
9. Once on`/verify/session/errors/warning` click Try again
10. You should be taken to `/verify/in_person/verify_info` (expected behavior).  You should not be taken to `/verify/document_capture` (not expected behavior).

##  🎥  Recordings
Before:

https://github.com/18F/identity-idp/assets/125507397/45737094-48dc-4f90-b847-700658c9b777

After: 

https://github.com/18F/identity-idp/assets/125507397/afc55339-cb92-4a30-bcfc-bf729b7b12f9
